### PR TITLE
Holtcl code review2

### PR DESCRIPTION
**Pull request by holtcl**
Analysis of Program:
This program is designed to retrieve compiled data from Covid-19 cases from across the globe and present them to the user in an organized fashion. Overall the program functioned well and appeared to be coded appropriately. Since the last code review they added functionality to the other fragments of the code. The code was very well compiled and I honestly had trouble finding things to critique within it.

Was the Project Available in Github on Time:
Yes

Is the program documented well enough for you to understand:
Yes, the project was very well documented throughout

Does the Program Compile:
Yes

Rational Behind Changes
Per Android Studio's suggestions I changed the text size references from dp to sp. I also added another fragment to bring the bar chart to full screen to allow for easier reading.

Technical Suggestions:
Try to ensure the text is easy to read in all parts of the app. I found the font very small within the bar chart, though it was understandable as the chart was fit onto the screen with other data.
Ensure that the proper values are used when designing the layout. Android Studio gave me warnings about "dp" being used when "sp" was recommended in the xml layout files.
Ensure to utilize unit testing. The only test I saw was an example test
 Keep up the good work. Your code is very well put together and well documented.

Technical Concepts I Learned:
How to make a specific object clickable
How to create a full screen display
The use of the News View and Item Adapter

**EDIT: Created a second pull request with the changes here:
https://github.com/sukhbirsekhon/covid19Tracker/pull/51/files
This one compiles properly and passes CircleCI test. While this one originally complied and ran fine something went wrong when I tried to merge it and it no longer worked. Though it looked like some of the code I had changed (changing the dp to sp) had already been fixed by the group**

### CODE REVIEW

File: app/src/main/java/edu/uc/groupproject/covid19tracker/BarChartFragment.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+         * If [AUTO_HIDE] is set, the number of milliseconds to wait after
+         * user interaction before hiding the system UI.
+         */
+        private const val AUTO_HIDE_DELAY_MILLIS = 3000
+
+        /**
+         * Some older devices needs a small delay between UI widget updates
+         * and a change of the status and navigation bar.
+         */
+        private const val UI_ANIMATION_DELAY = 300
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/BarChartFragment.kt:169]: Added new fragment to show bar chart on a larger screen when clicked on

File: app/src/main/java/edu/uc/groupproject/covid19tracker/MainActivity.kt [hunk 2]
```diff
@@ -10,7 +11,7 @@ import edu.uc.groupproject.covid19tracker.ui.main.MainViewModel
 import edu.uc.groupproject.covid19tracker.ui.main.NewsFragment
 
 class MainActivity : AppCompatActivity() {
-
+    lateinit var barChartFragment: BarChartFragment
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/MainActivity.kt:14]: added bar chart variable

File: app/src/main/java/edu/uc/groupproject/covid19tracker/MainActivity.kt [hunk 3]
```diff
@@ -57,4 +58,13 @@ class MainActivity : AppCompatActivity() {
...
     }
+
+    fun barChartFragment(view: View) {
+        barChartFragment = BarChartFragment()
+        supportFragmentManager
+            .beginTransaction()
+            .replace(R.id.container, barChartFragment)
+            .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
+            .commit()
+    }
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/MainActivity.kt:69]: added function to pull up bar chart fragment

File: app/src/main/res/layout/fragment_bar_chart.xml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                        android:text="@string/dummy_button" />
+
+                </LinearLayout>
+            </FrameLayout>
+        </com.github.mikephil.charting.charts.BarChart>
+
+    </androidx.cardview.widget.CardView>
+
+    <!-- This FrameLayout insets its children based on system windows using
+         android:fitsSystemWindows. -->
```
Comment (holtcl) [inline app/src/main/res/layout/fragment_bar_chart.xml:68]: Bar chart fragment layout

File: app/src/main/res/layout/main_fragment.xml [hunk 5]
```diff
@@ -138,7 +140,7 @@
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
                 android:textColor="#000"
-                android:textSize="16dp"
+                android:textSize="16sp"
```
Comment (holtcl) [inline app/src/main/res/layout/main_fragment.xml:143]: changed instances of textSize = dp to sp per Android Studio recommendations.

**Comment by holtcl** [comment]
Unsure why circleci tests failed? Program compiled and ran fine for me before commit.

**Comment by holtcl** [comment]
Created a second pull request with the changes here:
https://github.com/sukhbirsekhon/covid19Tracker/pull/51/files
 This one compiles properly and passes CircleCI test

File: app/src/main/java/edu/uc/groupproject/covid19tracker/BarChartFragment.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/groupproject/covid19tracker/BarChartFragment.kt:169]: Ah, cool!

**Comment by discospiff** [comment]
dp to sp looks like a good idea.
The new Bar Chart Fragment is new functionality, not necessarily a technical debt reduction, so I'll leave it to the group to decide whether or not to merge.

**Comment by sukhbirsekhon** [comment]
Closing this pull request since you already made new pull request

**Pull request closed without merging**